### PR TITLE
impr: upgrade spring version to 2.3.0

### PR DESCRIPTION
**Pull request by tomaszpowroznik**

### CODE REVIEW

File: service/src/main/resources/application.yaml [hunk 1]
```diff
@@ -8,7 +8,9 @@ management:
   endpoint:
     health:
       show-details: "always"
-
+  health:
+    probes:
+      enabled: false
```
Comment (tomaszpowroznik) [inline service/src/main/resources/application.yaml:13]: new 2.3.0 feature that collides with 
uk.gov.hmcts.reform.health-spring-boot-starter: '0.0.5'

File: service/src/main/java/uk/gov/hmcts/reform/fpl/config/LocalAuthorityCodeLookupConfiguration.java [hunk 2]
```diff
@@ -19,7 +20,7 @@ public LocalAuthorityCodeLookupConfiguration(@Value("${fpl.local_authority_email
     }
 
     public String getLocalAuthorityCode(String emailDomain) {
-        checkNotNull(emailDomain, "Email domain cannot be null");
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/config/LocalAuthorityCodeLookupConfiguration.java:22]: deprecated in assertj version required by spring 2.3.0, anyway it was from testing support used in prod code

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/config/LocalAuthorityCodeLookupConfiguration.java:22]: Maybe it was intended to use the [guava preconditions](https://github.com/google/guava/wiki/PreconditionsExplained), it has the same method name

File: service/src/main/resources/application.yaml (hunk 1 above)
Comment (lewisbirks) [reply service/src/main/resources/application.yaml:13]: There has been [some discussion](https://hmcts-reform.slack.com/archives/C8SR5CAMU/p1589978499323600) around this already, don't know if it is helpful

File: service/src/main/java/uk/gov/hmcts/reform/fpl/config/LocalAuthorityCodeLookupConfiguration.java (hunk 2 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/config/LocalAuthorityCodeLookupConfiguration.java:22]: could be, anyway prefer use native java counterpart

File: service/src/main/resources/application.yaml (hunk 1 above)
Comment (tomaszpowroznik) [reply service/src/main/resources/application.yaml:13]: good to know, was not aware about this discussion.

**Pull request closed without merging**